GeoLib/AnalyticalGeometry-impl.h

```diff
@@ -10,23 +10,23 @@
 namespace GeoLib
 {
 
-template <class T_POINT>
-void getNewellPlane (const std::vector<T_POINT*>& pnts,
+template <typename InputIterator>
+void getNewellPlane (InputIterator pnts_begin, InputIterator pnts_end,
                      MathLib::Vector3 &plane_normal,
                      double& d)
 {
     d = 0;
     MathLib::Vector3 centroid;
-    std::size_t n_pnts(pnts.size());
-    for (std::size_t i(n_pnts - 1), j(0); j < n_pnts; i = j, j++) {
-        plane_normal[0] += ((*(pnts[i]))[1] - (*(pnts[j]))[1])
-                           * ((*(pnts[i]))[2] + (*(pnts[j]))[2]); // projection on yz
-        plane_normal[1] += ((*(pnts[i]))[2] - (*(pnts[j]))[2])
-                           * ((*(pnts[i]))[0] + (*(pnts[j]))[0]); // projection on xz
-        plane_normal[2] += ((*(pnts[i]))[0] - (*(pnts[j]))[0])
-                           * ((*(pnts[i]))[1] + (*(pnts[j]))[1]); // projection on xy
-
-        centroid += *(pnts[j]);
+    std::size_t n_pnts(std::distance(pnts_begin, pnts_end));
```

Comment:
`n_pnts` are not used here but at the end of the function; please move down.
Also it is possible that std::distance returns a negative value causing underflow, or zero causing division by zero.
Suggest to use signed type and check for negative and zero values.